src/openms/include/OpenMS/ANALYSIS/ID/IDRipper.h

```diff
@@ -70,10 +70,14 @@ namespace OpenMS
       @param proteins Peptide identification annotated with file origin
       @param peptides Protein identification
     */
-    void rip(std::map<String, std::pair<std::vector<ProteinIdentification>, std::vector<PeptideIdentification> > > & ripped, std::vector<ProteinIdentification> & proteins, std::vector<PeptideIdentification> & peptides);
+    void rip(std::map<std::pair<UInt, UInt>, std::pair<std::vector<ProteinIdentification>, std::vector<PeptideIdentification> > > & ripped, std::vector<ProteinIdentification> & proteins, std::vector<PeptideIdentification> & peptides);
 
 private:
 
+    enum OriginAnnotationFormat { FILE_ORIGIN = 0, MAP_INDEX = 1, ID_MERGE_INDEX = 2, UNKNOWN = 3 };
```

Comment:
we usually go with `SIZE_OF_ORIGIN_ANNOTATION_FORMAT`, as last member so you can do
`std::array<std::string, SIZE_OF_ORIGIN_ANNOTATION_FORMAT> names_of_OriginAnnotationFormat = { ...};`

this makes iteration over the string representations easy... see below